Sofa/Component/Engine/Select/src/sofa/component/engine/select/BaseROI.inl

```diff
@@ -0,0 +1,779 @@
+                vertices.push_back(convertToVertex(DataTypes::getCPos(x0[t[(j + 1) % 4]])));
+            }
+
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[t[0]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[t[2]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[t[1]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[t[3]])));
+        }
+        vparams->drawTool()->drawLines(vertices, sizeFactor, color);
+    }
+
+    ///draw hexahedra in ROI
+    if( d_drawHexahedra.getValue())
+    {
+        std::vector<type::Vec3> vertices;
+        ReadAccessor< Data<vector<Hexa> > > hexahedraInROI = d_hexahedraInROI;
+        for (const auto& h : hexahedraInROI)
+        {
+            for (unsigned int j=0 ; j<8 ; j++)
+            {
+                vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[j]])));
+                vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[(j + 1) % 4]])));
+            }
+
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[0]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[2]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[1]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[3]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[4]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[5]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[6]])));
+            vertices.push_back(convertToVertex(DataTypes::getCPos(x0[h[7]])));
+        }
+        vparams->drawTool()->drawLines(vertices, sizeFactor, color);
+    }
+
+    vparams->drawTool()->restoreLastState();
+
+}
+
+template<typename DataTypes, typename Element>
+constexpr auto getCenter(const Element& e, const typename DataTypes::VecCoord& x0) -> typename DataTypes::CPos
+{
+    constexpr auto NumberOfNodes = Element::NumberOfNodes;
+
+    assert(NumberOfNodes > 0);
+
+    typename DataTypes::CPos center{};
+    for (const auto eid : e)
+    {
+        center += DataTypes::getCPos(x0[eid]);
+    }
+
+    center = center / static_cast<typename DataTypes::Real>(NumberOfNodes);
+
+    return center;
+}
+
+template<typename DataTypes, typename Element>
+bool isElementInROI(const Element& e, const typename DataTypes::VecCoord& x0, const std::function<bool(const typename DataTypes::CPos&)>& isPointInROI)
+{
+    const auto center = getCenter<DataTypes>(e, x0);
+
+    return isPointInROI(center);
+}
+
+template<typename DataTypes, typename Element>
+bool isElementInStrictROI(const Element& e, const typename DataTypes::VecCoord& x0, const std::function<bool(const typename DataTypes::CPos&)>& isPointInROI)
+{
+    for (const auto eid : e)
+    {
+        if (!isPointInROI(DataTypes::getCPos(x0[eid])))
+            return false;
+    }
+
+    return true;
```

Comment:
you could use `std::none_of`

Reply:
I used std::all_of (with the predicate isPointInRoi), makes it clearer to me